src/utils/hooks/useUpdateReviewAssignment.tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  useUpdateReviewAssignmentMutation,
+  ReviewAssignmentStatus,
+  ReviewAssignmentPatch,
+  TemplateElementCategory,
+  Trigger,
+} from '../generated/graphql'
+import { AssignmentDetailsNEW, FullStructure } from '../types'
+
+// below lines are used to get return type of the function that is returned by useUpdateReviewAssignmentMutation
+type UseUpdateReviewAssignmentMutationReturnType = ReturnType<
+  typeof useUpdateReviewAssignmentMutation
+>
+type PromiseReturnType = ReturnType<UseUpdateReviewAssignmentMutationReturnType[0]>
+// hook used to assign section/s to user as per type definition below (returns promise that resolve with mutation result data)
+type UseUpdateReviewAssignment = (
+  structure: FullStructure
+) => {
+  assignSectionToUser: (props: {
+    // Section code is optional if omitted all sections are assigned
+    sectionCode?: string
+    assignment: AssignmentDetailsNEW
+    // isSelfAssignment defaults to false
+    isSelfAssignment?: boolean
+  }) => PromiseReturnType
+}
+
+type ConstructAssignSectionPatch = (
+  sectionCode?: string,
+  isSelfAssignment?: boolean
+) => ReviewAssignmentPatch
+
+const useUpdateReviewAssignment: UseUpdateReviewAssignment = (structure) => {
+  const [updateAssignment] = useUpdateReviewAssignmentMutation()
+
+  const constructAssignSectionPatch: ConstructAssignSectionPatch = (
+    sectionCode,
+    isSelfAssignment
+  ) => {
+    const elements = Object.values(structure?.elementsById || {})
+
+    // filter elements by question category or by section code (if one passed through)
+    const assignableElements = elements.filter(
+      ({ element }) =>
+        (!sectionCode || element.sectionCode === sectionCode) &&
+        element.category === TemplateElementCategory.Question
+    )
+
+    const createReviewQuestionAssignments = assignableElements.map((element) => ({
+      templateElementId: element.element.id,
+    }))
+
+    return {
+      status: ReviewAssignmentStatus.Assigned,
+      // onReviewSelfAssign will trigger core action to disallow other reviewers to self assign
+      trigger: isSelfAssignment ? Trigger.OnReviewSelfAssign : Trigger.OnReviewAssign,
+      // TODO: change to timeUpdated
```

Comment:
Adding a placeholder issue for that

Reply:
Created a placeholder note for this issue [here](https://github.com/orgs/openmsupply/projects/6#card-58402523)